src/unix.rs

```diff
@@ -61,3 +61,27 @@ pub fn try_wait_without_reaping(handle: Handle) -> io::Result<bool> {
                            format!("unexpected si_signo from waitid: {}", siginfo.si_signo)))
     }
 }
+
+pub fn terminate(child: &super::SharedChild) -> io::Result<()> {
+    child.send_signal(libc::SIGTERM)
+}
+
+pub trait SharedChildExt {
+    /// Sends a signal to the child process with `libc::kill`.
+    fn send_signal(&self, signal: libc::c_int) -> io::Result<()>;
+}
+
+impl SharedChildExt for super::SharedChild {
+    fn send_signal(&self, signal: libc::c_int) -> io::Result<()> {
+        let status = self.state_lock.lock().unwrap();
```

Comment:
Could you add a comment, that holding this lock while we signal below is important for preventing the PID race?  Similar to the comment in `kill`.